### DISCUSSION TITLE
Markdown in "git tag" message not rendered in Release

### DISCUSSION THREAD

**Discussion by ahmetb** [community/community#22611] [New to GitHub]
<p>Hi folks, I’m tagging my annotated git releases as follows:</p>
<pre><code>git tag -a v0.4 -m "$(cat RELNOTES.md)"
</code></pre>
<p>The RELNOTES.md actually has a markdown, listing tag header + message body:</p>
<pre><code>krew v0.3

Installation
------------
To install this release, refer to the [installation instructions](https://www.example.com)

Acknowledgements
----------------
Thanks to contributors for helping out in this release:
- The Dog
- Good cat
</code></pre>
<p>So when I push the tag to my repo, GitHub doesn’t automatically create a release (expected).</p>
<p>What’s weird is that when I go to Releases --&gt; Tags of my repository and click  <strong>“Create Release”</strong> , I’m presented with a form with empty release header + message (unexpected): </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/32b978799d5f447034394ed4cb7136a94458055f.png" data-download-href="/uploads/short-url/7eJf21BrBziLw0nCZBJlSdN4qM7.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/159209/181072182-00964af7-0d72-4c57-bd80-160cfe5b930c.png" alt="image.png" data-base62-sha1="7eJf21BrBziLw0nCZBJlSdN4qM7" width="560" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/32b978799d5f447034394ed4cb7136a94458055f_2_560x500.png, https://aws1.discourse-cdn.com/github/optimized/1X/32b978799d5f447034394ed4cb7136a94458055f_2_840x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/32b978799d5f447034394ed4cb7136a94458055f_2_1120x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/32b978799d5f447034394ed4cb7136a94458055f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1526×1362 123 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Next, I click <strong>Publish Release</strong> , without filling out these fields, and my release is created. However, the release body is not shown as Markdown (unexpected), but plaintext: </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/79bafefe9be1a1ed868c1e0a25e87e45781577b1.png" data-download-href="/uploads/short-url/hmSpuvpPSpPQllgr6LsPUsDbvO1.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/159209/181072183-4677654d-c85b-4997-8010-6b303197b6ea.png" alt="image.png" data-base62-sha1="hmSpuvpPSpPQllgr6LsPUsDbvO1" width="690" height="399" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/79bafefe9be1a1ed868c1e0a25e87e45781577b1_2_690x399.png, https://aws1.discourse-cdn.com/github/optimized/1X/79bafefe9be1a1ed868c1e0a25e87e45781577b1_2_1035x598.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/79bafefe9be1a1ed868c1e0a25e87e45781577b1_2_1380x798.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/79bafefe9be1a1ed868c1e0a25e87e45781577b1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1700×984 70.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Now if I click “Edit Release” and copy/paste the same markdown tag message into the Release message, it renders Markdown as expected.</p>
<p>So my question is, how are people supposed to convert annotated git tag messages into GitHub Release message body without doing manual work?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>The process that you’re using copies the contents of a Markdown file into the text attribute of an annotated Git tag. There’s nothing that indicates to GitHub that the contents of that text string are anything other than plain text. So yes, if you do nothing, GitHub uses the annotation as the release notes as plain text. But if you edit the release notes in a field that explicitly supports Markdown, it assumes that you’re using Markdown and does the proper translation.</p>
<p>If you want to automate the process of having a GitHub release object with Markdown release notes, you could write a script to <a href="https://developer.github.com/v3/repos/releases/#create-a-release" rel="nofollow noopener">create the release object storing the release notes in the <code>body</code> field</a> via the GitHub API.</p>
<p>I hope that helps!</p>
